history-service/controllers/history-controller.js

```diff
@@ -1,43 +1,19 @@
-const pool = require("../psql-db.js");
 const express = require("express");
 const router = express.Router();
 
+const history = require("../models/history-model.js");
+
 async function saveAttempt(req, res) {
     let { username, collaborated, title, qnId, difficulty, language, attempt, date } = req.body;
 
-    console.log ("save!");
-    pool.query(
-        `INSERT INTO history (username, collaborated, title, qnid, difficulty, language, attempt, date)
-        VALUES ($1, $2, $3, $4, $5, $6, $7, $8)`,[username, collaborated, title, qnId, difficulty, language, attempt, date])
-        .then(response =>
-            console.log("saving attempt to the database successfully"))
-        .catch(error =>
-            console.log("Error saving attempts to database", error));
+    const result = await history.saveAttempt(username, collaborated, title, qnId, difficulty, language, attempt, date);
```

Comment:
I need error handling for the tests ;; like a return res.status(200).json({...}) for pass and res.status(...).json({...}) for errors.

Reply:
also just to let you know but for the tests ill add a deleteHistory command so i can test the save then delete the entry